### PR TITLE
gitea: Move from 'extras'

### PR DESCRIPTION
**Pull request by niheaven**
Ref: https://github.com/lukesampson/scoop-extras/pull/2660

### CODE REVIEW

File: bucket/gitea.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            "hash": "a38be091b407b7ade13fa637ea71588ca58c5afebd222c95fd1ba27630e08c47"
+        },
+        "32bit": {
+            "url": "https://dl.gitea.io/gitea/1.9/gitea-1.9-windows-4.0-386.exe#/gitea.exe",
+            "hash": "81ff3c007674c41d77653d6d0f5a2da4aa09d42f2bb818f4ab3c49d3ac16481d"
+        }
+    },
+    "post_install": [
+        "echo 'Creating $persist_dir\\custom\\conf.app.ini'",
+        "if(!(test-path \"$persist_dir\\custom\\conf\")) { mkdir \"$persist_dir\\custom\\conf\" > $null }",
```
Comment (Ash258) [inline bucket/gitea.json]: Do not use alias and output redirection

File: bucket/gitea.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        "if(!(test-path \"$persist_dir\\custom\\conf\\app.ini\")) {",
+        "    $gitea = \"[database]`nPATH = $persist_dir/data/gitea.db\"",
```
Comment (Ash258) [inline bucket/gitea.json]: Use array please

File: bucket/gitea.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        "    $gitea += \"`n[repository]`nROOT = $persist_dir/repositories\"",
+        "    $gitea += \"`n[log]`nROOT_PATH = $persist_dir/log\"",
+        "    echo $gitea | % {$_.replace(\"\\\",\"/\") } | out-file -encoding oem \"$persist_dir\\custom\\conf\\app.ini\"",
```
Comment (Ash258) [inline bucket/gitea.json]: - Fix encoding
- Replace could be called on array without foreach
- Simple `Set-Content ($gitea -replace) ..app.ini -Encoding ASCII` is good

File: bucket/gitea.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            "url": "https://dl.gitea.io/gitea/1.9/gitea-1.9-windows-4.0-amd64.exe#/gitea.exe",
+            "hash": "a38be091b407b7ade13fa637ea71588ca58c5afebd222c95fd1ba27630e08c47"
+        },
+        "32bit": {
+            "url": "https://dl.gitea.io/gitea/1.9/gitea-1.9-windows-4.0-386.exe#/gitea.exe",
+            "hash": "81ff3c007674c41d77653d6d0f5a2da4aa09d42f2bb818f4ab3c49d3ac16481d"
+        }
+    },
+    "post_install": [
+        "echo 'Creating $persist_dir\\custom\\conf.app.ini'",
```
Comment (Ash258) [inline bucket/gitea.json]: Alias

File: bucket/gitea.json [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        "64bit": {
+            "url": "https://dl.gitea.io/gitea/1.9/gitea-1.9-windows-4.0-amd64.exe#/gitea.exe",
+            "hash": "a38be091b407b7ade13fa637ea71588ca58c5afebd222c95fd1ba27630e08c47"
+        },
+        "32bit": {
+            "url": "https://dl.gitea.io/gitea/1.9/gitea-1.9-windows-4.0-386.exe#/gitea.exe",
+            "hash": "81ff3c007674c41d77653d6d0f5a2da4aa09d42f2bb818f4ab3c49d3ac16481d"
+        }
+    },
+    "post_install": [
```
Comment (Ash258) [inline bucket/gitea.json]: `post_install` is editiong persist_dir, but there is no `persist` property.

File: bucket/gitea.json [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        "if(!(test-path \"$persist_dir\\custom\\conf\")) { mkdir \"$persist_dir\\custom\\conf\" > $null }",
+        "if(!(test-path \"$persist_dir\\custom\\conf\\app.ini\")) {",
+        "    $gitea = \"[database]`nPATH = $persist_dir/data/gitea.db\"",
+        "    $gitea += \"`n[repository]`nROOT = $persist_dir/repositories\"",
+        "    $gitea += \"`n[log]`nROOT_PATH = $persist_dir/log\"",
+        "    echo $gitea | % {$_.replace(\"\\\",\"/\") } | out-file -encoding oem \"$persist_dir\\custom\\conf\\app.ini\"",
+        "}"
+    ],
+    "bin": "gitea.exe",
+    "notes": "Run 'gitea web -c \"$persist_dir\\custom\\conf\\app.ini\"' to start the webserver",
```
Comment (Ash258) [inline bucket/gitea.json]: Could this be added as binary? `gitea-webserver`

**Comment by niheaven** [comment]
I'll check it again. Just move manifest here.

**Comment by niheaven** [comment]
Really want to know who create this manifest...

**Comment by r15ch13** [comment]
@niheaven me :grin:
https://github.com/lukesampson/scoop-extras/commit/a5385499ab8c52455dd91c2268dbb1f8df2d53d8

**Pull request merged**